src/TTImp/Elab/Case.idr

```diff
@@ -460,7 +460,9 @@ checkCase rig elabinfo nest env fc opts scr scrty_in alts exp
         = case getFn x of
                IVar _ n =>
                   do defs <- get Ctxt
-                     [(n', (_, ty))] <- lookupTyName n (gamma defs)
+                     [(n', (_, ty))] <- do nested <- maybe (pure []) (\n => lookupTyName n (gamma defs))
+                                                      (join $ Builtin.fst <$> List.lookup n nest.names)
+                                           (nested ++) <$> lookupTyName n (gamma defs)
```

Comment:
Is it worth lifting this pattern to a top-level definition
```
lookupLocalTyNames : {auto c : Ref Ctxt Defs} -> NestedNames vars -> Name -> Core (List (Name, NF []))
```
? Surely this pattern occurs in other places too?

Reply:
Yes please. Whatever the common pattern is should be isolated so that it is used uniformly!